File: examples/python/py/a_SimpleDomains.py
# -*- coding: utf-8 -*-
"""
Simple domains example
"""

import matplotlib.pyplot as plt
import sys, os
sys.path.append(os.path.relpath("..//..//..//src//python"))
import hiped as hp

#%% 1) 0D domain

domain0 = hp.Domain(1) # a single dot
plt.clf()
domain0.plot()
plt.show()

#%% 2) 1D domain

domain1 = hp.Domain(2) # a 1D line segment has 2 vertices
plt.figure()
domain1.plot()
plt.show()

#%% 3) 2D domain

domain2 = hp.Domain(5) # a regular pentagon
plt.figure()
domain2.plot()
plt.show()

#%% 4) 3D domain

domain3 = hp.Domain("diamond6") # available : "tetra", "cube", "diamondN", "prismN" (N being an integer)
plt.figure()
domain3.plot()
plt.show()

File: examples/matlab/mfiles/a_SimpleDomains.m
%% Simple domains examples

clear ; close ; path=pwd();
[s,e]=regexp(path,'.+(?=[\\/]{1,2}examples)');
path=path(s:e); addpath(genpath(path)); clear s e path
%% 1) 0D domain

domain0D = Domain(1); % a single dot
figure()
domain0D.plot()
%% 1) 1D domain

domain1D = Domain(2); % a 1D line segment has 2 vertices
figure()
domain1D.plot()
%% 2) 2D domain

nVertices2D = 5;
domain2D = Domain(nVertices2D);  % a regular polygon
figure()
domain2D.plot()
%% 3) 3D domain

name3D = "diamond8"; % (supported: tetra, cube, diamond, prism, pyramid)
domain3D = Domain(name3D);
figure()
domain3D.plot()